addons/http_routing/models/ir_http.py

```diff
@@ -22,8 +22,8 @@
 _logger = logging.getLogger(__name__)
 
 # NOTE: the second pattern is used for the ModelConverter, do not use nor flags nor groups
-_UNSLUG_RE = re.compile(r'(?:(\w{1,2}|\w[A-Za-z0-9-_]+?\w)-)?(-?\d+)(?=$|\/|#|\?)')
-_UNSLUG_ROUTE_PATTERN = r'(?:(?:\w{1,2}|\w[A-Za-z0-9-_]+?\w)-)?(?:-?\d+)(?=$|\/|#|\?)'
+_UNSLUG_RE = re.compile(r'(?:(\w{1,2}|\w[\w\-_]+?\w)-)?(-?\d+)(?=$|\/|#|\?)')
+_UNSLUG_ROUTE_PATTERN = r'(?:(?:\w{1,2}|\w[\w\-_]+?\w)-)?(?:-?\d+)(?=$|\/|#|\?)'
```

Comment:
Nitpicking, the way git works, it is going to save a new file with the new code. Since the only change to the file is `A-Za-z0-9` -> `\w` which doesn't change anything, I would argue we shouldn't be changing this file.

Also FWI `\w` is `A-Za-z0-9_`, and you don't need to escape `-` when it is the first/last char inside the `[]` so it could have been rewritten as `[\w-]`